### PR TITLE
Use structopt for command parsing

### PR DESCRIPTION
**Pull request by Jokler**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -169,60 +170,63 @@ impl Application {
 
     async fn on_text(&self, message: Message) -> Result<(), AudioPlayerError> {
         let msg = message.text;
-        if msg.starts_with("!") {
-            let tokens = msg[1..].split_whitespace().collect::<Vec<_>>();
+        if msg.starts_with("poke") {
```
Comment (fkaa) [inline src/main.rs:173]: pokeadd <url>? what is the syntax now?

File: src/main.rs [hunk 2]
```diff
@@ -169,60 +170,63 @@ impl Application {
...
+            let tokens = msg.split_whitespace().collect::<Vec<_>>();
+
+            let args = Command::from_iter_safe(&tokens);
+            match args {
+                Ok(v) => self.on_command(v).await?,
+                Err(e) => self.send_message(&format!("\n{}", e.message)),
```
Comment (fkaa) [inline src/main.rs:179]: `error!` log?

**Review by fkaa — CHANGES_REQUESTED** [review]

File: src/main.rs (hunk 2 above)
Comment (Jokler) [reply src/main.rs:179]: An error over here means that the user entered an invalid command, I don't think that user errors have to be logged.

File: src/main.rs (hunk 1 above)
Comment (Jokler) [reply src/main.rs:173]: It's `poke add <url>` to add to add now.
Anything that starts with poke works though so you could type pokebot too if you wanted to.

**Review by fkaa — APPROVED** [review]

**Pull request merged**